<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT<h1>
<hr><h3>Analyst : Jeremy Bryan T. Villanueva</h3>

In [63]:
#import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Converting Database(s) into DataFrame</h3>
<h2>Requirements:</h2>
<ul>
    <li>Name of the DataFrame</li>
    <li>File Path</li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Describe Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h3>Checking for Null Values</h3>

In [5]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filtering a column</h3>

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [10]:
penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [11]:
penguins[penguins['species'] == 'Gentoo']

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [12]:
penguins[penguins['species'] != 'Gentoo']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[220 rows x 8 columns]

In [13]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [14]:
adelie = penguins[penguins['species'] == 'Adelie']
adelie

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [15]:
gentoo = penguins[penguins['species'] == 'Gentoo']
gentoo

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h2>Combining Table</h2>

<h3>How many of these chinstrap species weight is above 4000g?</h3>

In [16]:
(penguins['species'] == 'Chinstrap')

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [17]:
(penguins['body_mass_g'] >= 40000

_IncompleteInputError: incomplete input (3711948209.py, line 1)

In [18]:
(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [19]:
chinstrap_weight = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]
chinstrap_weight

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [20]:
chinstrap_weight.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h2>FILLING UP NULL VALUES</h2>

In [21]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [22]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [23]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [24]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [25]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [26]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [27]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [28]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [29]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [30]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [31]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [32]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [33]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [34]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [35]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [36]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>CHANGING THE DATA TYPE</h2>

In [37]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [38]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [39]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


penguins.describe()

<h2>HOW TO CREATE YOUR OWN COLUMN</h2>

In [41]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h2>Data Aggreation</h2>
<h3>How many species are live in every islands?</h3>

In [67]:
species_island = penguins.groupby(['species','island']).count()
species_island

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [68]:
species_island = penguins.groupby(['species','island','sex']).size()
species_island

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h3>Average weight of every species in island and sex?</h3>

In [70]:
avg_island_sex = penguins.groupby(['species','island','sex'])['body_mass_g'].mean()
avg_island_sex

species    island     sex   
Adelie     Biscoe     female    3369.318182
                      male      4050.000000
           Dream      female    3344.444444
                      male      4008.620690
           Torgersen  female    3395.833333
                      male      3990.241228
Chinstrap  Biscoe     female            NaN
                      male              NaN
           Dream      female    3527.205882
                      male      3938.970588
           Torgersen  female            NaN
                      male              NaN
Gentoo     Biscoe     female    4679.741379
                      male      5411.011430
           Dream      female            NaN
                      male              NaN
           Torgersen  female            NaN
                      male              NaN
Name: body_mass_g, dtype: float64

<h2>CORRELATION ANALYSIS</h2>
<hr>
<h3>Testing Correlation using Pearson R</h3>
<hr>
<strong>Pearson R</strong> can test cocrrelation between two Variables, given that both Variables are <strong>NUMERICAL DATA</strong>

In [96]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])
flipper_mass

np.float64(0.8712017673060114)

In [98]:
bill_mass = penguins['body_mass_g'].corr(penguins['bill_length_mm'])
bill_mass

np.float64(0.5951098244376305)

<h2>Categorical Data</h2>
<hr>
<h3>Boday Mass V sex</h3>

In [100]:
from scipy.stats import f_oneway

In [101]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [102]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [104]:
stats, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] == 'female']['body_mass_g'])
p_value

np.float64(3.0226600487050246e-15)

<h2>Exploratory Data Analytics</h2>
<hr>

<h3>Question 1</h3>
<h4>Which species has the lightest body mass?</h4>

In [42]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [43]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df['body_mass_g'].mean()

np.float64(5068.965761177136)

In [44]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [45]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df['body_mass_g'].mean()

np.float64(3703.95891043398)

<h3>Insight No. 1</h3>
<h4>Adelie has the lightiest body mass in all three(3) species.</h4>

<h2>Question 2</h2>
<h3>Which species has the longest bill length?</h3>

In [46]:
penguins['bill_length_mm'].mean()

np.float64(43.92192982456141)

In [47]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df['bill_length_mm'].mean()

np.float64(47.475983305036785)

In [48]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [49]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df['bill_length_mm'].mean()

np.float64(38.82514427516159)

<h3>Insight No.2</h3>
<h4>In all three(3) species Chinstrap has the longest Bill length.</h4>

<h2>Question 3</h2>
<h3>Which island has the heaviest penguins?</h3>

In [50]:
penguins['mass_kg'].mean()

np.float64(4.2020348837209305)

In [51]:
torgersen_df = penguins[penguins['island'] == 'Torgersen']
torgersen_df['mass_kg'].mean()

np.float64(3.7115384615384617)

In [52]:
biscoe_df = penguins[penguins['island'] == 'Biscoe']
biscoe_df['mass_kg'].mean()

np.float64(4.714880952380952)

In [53]:
dream_df = penguins[penguins['island'] == 'Dream']
dream_df['mass_kg'].mean()

np.float64(3.7129032258064516)

<h3>Insight No. 3</h3>
<h4>The heaviest penguins will find in Biscoe island.</h4>

<h2>Question 4</h2>
<h3>Which gender is dominant in Adélie penguins?</h3>

In [54]:
adelie_sex = penguins[(penguins['species'] == 'Adelie') & (penguins['sex'] == 'male')]
adelie_sex.value_counts()

species  island     bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex   year  mass_kg
Adelie   Biscoe     37.60000        19.10000       194.000000         3750.000000  male  2008  3.8        1
                    37.70000        18.70000       180.000000         3600.000000  male  2007  3.6        1
                    37.80000        20.00000       190.000000         4250.000000  male  2009  4.2        1
                    38.20000        18.10000       185.000000         3950.000000  male  2007  4.0        1
                                    20.00000       190.000000         3900.000000  male  2009  3.9        1
                                                                                                         ..
         Torgersen  43.10000        19.20000       197.000000         3500.000000  male  2009  3.5        1
                    43.92193        17.15117       200.915205         4201.754386  male  2007  4.2        1
                    44.10000     

In [55]:
adelie_sex = penguins[(penguins['species'] == 'Adelie') & (penguins['sex'] == 'female')]
adelie_sex.value_counts()

species  island     bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex     year  mass_kg
Adelie   Biscoe     34.5            18.1           187.0              2900.0       female  2008  2.9        1
                    35.0            17.9           190.0              3450.0       female  2008  3.4        1
                                                   192.0              3725.0       female  2009  3.7        1
                    35.3            18.9           187.0              3800.0       female  2007  3.8        1
                    35.5            16.2           195.0              3350.0       female  2008  3.4        1
                                                                                                           ..
         Torgersen  39.6            17.2           196.0              3550.0       female  2008  3.6        1
                    40.2            17.0           176.0              3450.0       female  2009  3.4        1
               

<h3>Insight No.4</h3>
<h4>The dominant gender in Adelie specie is male.</h4>

<h2>Question 5</h2>
<h3>How many penguins were on Dream Island?</h3>

In [95]:
avg_island = penguins.groupby(['species','island']).size()
avg_island

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

<h3>Insights No.5</h3>
<h4>There are 56 adelie penguins live in Dream island. There are 68 Chinstrap live in Dram island and 0 gentoo live in dream island. In total there 124 penguins live in Dream island.</h4>

<h2>Question 6</h2>
<h3>How many penguins were on Dream Island?</h3>